src/components/Asset/AssetActions/Trade/Swap.tsx

```diff
@@ -56,148 +61,189 @@ export default function Swap({
   const [spotPrice, setSpotPrice] = useState<string>()
   const [totalValue, setTotalValue] = useState<string>()
   const [tokenAmount, setTokenAmount] = useState<string>()
+  const [swapFee, setSwapFee] = useState<string>()
 
   useEffect(() => {
-    if (!asset || !balance || !values?.type) return
+    if (!assetExtended || !balance || !values?.type) return
```

Comment:
looking through effect I would assume it should `return` also when no `web3` available